src/Core/Grid/LinkBlockGridFactory.php

```diff
@@ -86,6 +95,11 @@ public function getGrids(array $hooks, array $filtersParams)
         foreach ($hooks as $hook) {
             $hookParams = $filtersParams;
             $hookParams['filters']['id_hook'] = $hook['id_hook'];
+            $hookParams['filters']['id_shop'] = [$this->shopContext->getContextShopID()];
+            
+            if (!$this->shopContext->isSingleShopContext()) {
+                $hookParams['filters']['id_shop'] = $this->shopContext->getContextListShopID();
```

Comment:
Why not  getContextListShopID ?

Reply:
Did you mean another method? `getContextListShopID` is the method he uses here 🤔

Reply:
@matthieu-rolland I think Pierre suggest to use only one method `getContextListShopID` which - indeed - should be enough, I'll check on that later